docker-compose-net/docker-compose-net.md

```diff
@@ -0,0 +1,234 @@
+
+For other operating systems, please install and configure the following separately before proceeding:
+
+  - [`docker-machine`](http://docs.docker.com/machine/#installation) binary (_`>= 0.8.0`_)
+  - [`docker`](https://docs.docker.com/installation/#installation) binary, at least the client (_`>= v1.6.x`_)
+  - [VirtualBox](https://www.virtualbox.org/wiki/Downloads) (_`>= 4.3.x`_)
+  - `curl` (_any version_)
+
+##<a name="attaching-containers"></a>Docker Compose and Attaching Containers to a Weave Network 
+ 
+ There are two different ways that you can attach containers to a Weave network when using Compose (which method to use is 
+ entirely up to you):
+ 
+ **1.** The Weave Net Docker API Proxy. See [Integrating Docker via the API Proxy](https://www.weave.works/docs/net/latest/weave-docker-api/).  
+ 
+ **2.**  The Docker Network Plugin framework. The Docker Network Plugin is used when 
+ Docker containers are started with the --net flag, for example: 
+ 
+ `docker run --net <docker-run-options>`
+ 
+ **Where,** 
+ 
+  * `<docker-run-options>` are the [docker run options](https://docs.docker.com/engine/reference/run/) 
+  you give to your container on start 
+ 
+ >>**Note:** If a Docker container is started with the `--net` flag, then the Weave Docker API Proxy
+ is automatically disabled and is not used to attach containers. See [Integrating Docker via the Network Plugin](https://www.weave.works/docs/net/latest/features/#plugin).
+
+
+##<a name="setup-proxy"></a>Docker Compose and the Weave Net Docker API Proxy
+
+
+The following is a simple setup that shows how to configure docker-compose for Weave Net across two hosts using the Docker API proxy. You will launch the alpine using Docker Compose and then run 'ping' to test the connectivity between the two hosts. 
+
+**1.** Create two hosts on Virtualbox with docker machine in two different terminal windows: 
+
+~~~bash
+docker-machine create -d virtualbox weave-compose-01
+~~~
+
+And, in a new terminal window, create the other VM: 
+
+~~~bash
+docker-machine create -d virtualbox weave-compose-02
+~~~
+
+**2.** In order to peer hosts on Weave Net, the IP address of one of the VMs you just created must be known beforehand: 
+
+~~~bash
+docker-machine ip weave-compose-02
+~~~
+
+Make a note of the IP address that appears. 
+
+**3.** Install and then launch Weave Net onto both VMs. From one terminal windown launch Weave: 
+
+~~~bash
+sudo curl -L git.io/weave -o /usr/local/bin/weave
+sudo chmod +x /usr/local/bin/weave
+weave launch
+~~~
+
+And in a second terminal launch Weave onto the other host, and pass the IP of the first host to it: 
+
+~~~bash
+sudo curl -L git.io/weave -o /usr/local/bin/weave
+sudo chmod +x /usr/local/bin/weave
+weave launch <ip address host-02>
+~~~
+
+**4.** Setup the environment to run the Weave Docker API proxy: 
+
+~~~bash
+eval $(weave env)
```

Comment:
Where is the user expected to run this? On one of the VMs (or both of them) or on their workstation?